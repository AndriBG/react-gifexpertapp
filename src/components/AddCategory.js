import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [ inputValue, setInputValue ] = useState('');

    const handleInputValue = (e) => {
        setInputValue(e.target.value)
    }

    const handleSubmit = (e) =>  {
        e.preventDefault();

        if ( inputValue.trim().length > 1 ) {
            setCategories(catgs => [inputValue, ...catgs]);
            setInputValue('');
        }

    }

    return (
        <form onSubmit= { handleSubmit }>
            {/* <p>{ inputValue }</p> */}
            <input
                type = 'Text'
                placeholder = 'Buscar'
                value = { inputValue }
                onChange = { handleInputValue }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
