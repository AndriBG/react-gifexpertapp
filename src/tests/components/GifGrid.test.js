import React from 'react';
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {

    let wrapper = shallow(<GifGrid category = { "Boruto" } />);

    test('<GifGrid /> debe mostrarse correctamente', () => {

        useFetchGifs.mockReturn({
            data: [],
            loading: true
        });

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imágenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturn({
            data: gifs,
            loading: false
        });

        // expect( wrapper ).toMatchSnapshot();

        // este es el texto que dice: "cargando...", que no debería de existir si hay gifs
        expect( wrapper.find('p').exists() ).toBe( false );

        // Este el componente GifGridItem que debería de haber si hay gifs
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });

});