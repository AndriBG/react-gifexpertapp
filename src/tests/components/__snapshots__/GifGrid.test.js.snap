// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> <GifGrid /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    Boruto
  </h3>
  cargando...
  <div
    className="card-grid"
  />
</Fragment>
`;
